brave/brave-core
# Offer google translate installation in translate bubbles

Fix https://github.com/brave/brave-browser/issues/5561

Below is the gif of how this PR looks like and it's easier to go through each commit when reviewing.
![gtranslate](https://user-images.githubusercontent.com/4730197/62578460-07daf880-b856-11e9-845e-5748e616d16f.gif)

This PR repurposed TranslateBubbleView and TranslateIconView to offer google translate installation in translate bubbles when `ENABLE_BRAVE_TRANSLATE` (go-translate) is disabled.
This UI is expected to be used in all desktop channels, and the prompt of translate bubbles could be disabled through language settings by toggling `Offer to translate pages that aren't in a language you read` as below.
<img width="706" alt="Screen Shot 2019-08-06 at 11 04 26 AM" src="https://user-images.githubusercontent.com/4730197/62578441-00b3ea80-b856-11e9-9a1b-764a6650fea9.png">

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
1. Open brave://settings
2. Toggle off the `Offer to translate pages that aren't in a language you read` setting in Languages -> Language.
3. Visit https://www.deutschland.de/de, translate bubble should not be shown.
4. Right click and `Translate to _your_language_` should not be shown in the context menu.
5. Toggle on the `Offer to translate pages that aren't in a language you read` setting in Languages -> Language.
6. Visit https://www.deutschland.de/de, the translate bubble should be shown to offer google translate installation.
7. After installation, both the icon & bubble in the location bar should be hidden and the page could be translated through Google Translate extension.
8. Open a new tab and visit https://www.deutschland.de/de again, translate bubble should not be shown.

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
